ansible/roles/master/templates/kube-api-hyperkube.yml.j2

```diff
@@ -0,0 +1,106 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: kube-apiserver
+spec:
+  hostNetwork: true
+  containers:
+  - name: kube-apiserver
+    image: gcr.io/google_containers/hyperkube:{{ hyperkube_version }}
+    args:
+    - /hyperkube 
+    - apiserver
+    -  "--bind-address={{ kube_apiserver_bind_address }}"
+    -  "--insecure-bind-address=127.0.0.1"
+    -  "--etcd-servers={% for node in groups['etcd'] %}http://{{ hostvars[node]['ansible_' + etcd_interface].ipv4.address }}:2379{% if not loop.last %},{% endif %}{% endfor %}"
+    -  "--cloud-provider="   
+    -  "--admission-control={{ admission_controllers }}"
+    -  "--service-cluster-ip-range={{ kube_service_addresses }}" 
+    -  "--client-ca-file=/etc/kubernetes/certs/ca.crt"
+    -  "--client-ca-file={{ kube_cert_dir }}/ca.crt"
+    -  "--tls-cert-file={{ kube_cert_dir }}/server.crt"
+    -  "--tls-private-key-file={{ kube_cert_dir }}/server.key"
+    -  "--secure-port={{ kube_master_api_port }}"
+    -  "--token-auth-file={{ kube_token_dir }}/known_tokens.csv"
+    -  "--v=2"
+    -  "--service-account-key-file={{ kube_cert_dir }}/server.crt"
+    -  "--allow-privileged=False"
+    -  "--log-dir=/var/log/kube-apiserver.log"     
+ 
+    ports:
+    - containerPort: 443
+      hostPort: 443
+      name: https
+    - containerPort: 7080
+      hostPort: 7080
+      name: http
+    - containerPort: 8080
+      hostPort: 8080
+      name: local
+    volumeMounts:
+    - mountPath: {{ kube_cert_dir }}
+      name: srvkube
+      readOnly: true
+    - mountPath: {{ kube_token_dir }}
+      name: kubetoken
+      readOnly: true
+    - mountPath: /var/log/kube-apiserver.log
+      name: logfile
+    - mountPath: /etc/ssl
+      name: etcssl
+      readOnly: true
+#    - mountPath: /usr/share/ssl
+#      name: usrsharessl
+#      readOnly: true
+    - mountPath: /var/ssl
+      name: varssl
+      readOnly: true
+#    - mountPath: /usr/ssl
+#      name: usrssl
+#      readOnly: true
+#    - mountPath: /usr/lib/ssl
+#      name: usrlibssl
+#      readOnly: true
+#    - mountPath: /usr/local/openssl
```

Comment:
I would figure these commented lines are needed to generate ssl certs, kets, etc..

If your WIP patch uses clear text between haproxy and kube-api, I think eparis will block the pr. I suggest not terminating ssl on haproxy, unless you can reencrypt on the backend.

I have not dug into the config, but contrib/service-loadbalancer uses haproxy and works with all the kube cluster addons:

https://github.com/kubernetes/contrib/blob/master/service-loadbalancer/template.cfg